run/run.go

```diff
@@ -12,39 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package sdk
+package run
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"os"
 
-	"github.com/conduitio/conduit-processor-sdk/internal"
-	"github.com/goccy/go-json"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit-processor-sdk/internal/wasm"
 )
 
 // Run is the 'entry point' for a processor. It runs a
 // 'get a command, send a reply' loop through which it
 // communicates with Conduit.
 //
 // A processor plugin needs to call this function in its main() function.
-func Run(p Processor) {
+func Run(p sdk.Processor) {
 	for {
-		cmd, err := internal.NextCommand()
+		cmd, err := wasm.NextCommand()
 		if err != nil {
-			fmt.Printf("failed retrieving next command: %v", cmd)
-			return
+			_, _ = fmt.Fprintf(os.Stderr, "failed retrieving next command: %v", err)
+			exitCode := 1
+			if errors.Is(err, sdk.ErrNoMoreCommands) {
+				exitCode = 0
+			}
+			os.Exit(exitCode)
```

Comment:
`os.Exit` is quite harsh:
> The program terminates immediately; deferred functions are not run.

Wouldn't it be better to just `return`?

Reply:
The intention behind os.Exit was to communicate the reason of the exit (graceful vs non-graceful etc.) Currently it's not being checked, we can have a simple return as well.

Reply:
I kept os.Exit, I think your reasoning makes sense. I added docs on top of `Run` to explicitly say how it's supposed to be used, so users don't do anything special there.